services/apps/alcs/src/alcs/search/notification/notification-advanced-search.service.spec.ts

```diff
@@ -73,63 +74,88 @@ describe('NotificationAdvancedSearchService', () => {
           provide: getRepositoryToken(LocalGovernment),
           useValue: mockLocalGovernmentRepository,
         },
+        {
+          provide: getRepositoryToken(Notification),
+          useValue: mockNotificationRepo,
+        },
+        {
+          provide: getRepositoryToken(NotificationSubmission),
+          useValue: mockNotificationSubmissionRepo,
+        },
+        {
+          provide: RedisService,
+          useValue: mockRedisService,
+        },
       ],
     }).compile();
 
     service = module.get<NotificationAdvancedSearchService>(
       NotificationAdvancedSearchService,
     );
 
+    mockNotificationSubmissionSearchViewRepository.createQueryBuilder.mockReturnValue(
+      mockQuery as any,
+    );
+
     mockLocalGovernmentRepository.findOneByOrFail.mockResolvedValue(
       new LocalGovernment(),
     );
+
+    mockRedisService.getClient.mockReturnValue({
+      get: async () => null,
+      setEx: async () => null,
+    } as any);
   });
 
   it('should be defined', () => {
     expect(service).toBeDefined();
   });
 
   it('should successfully build a query using all search parameters defined', async () => {
-    mockNotificationSubmissionSearchViewRepository.createQueryBuilder.mockReturnValue(
-      mockQuery as any,
+    mockNotificationRepo.find.mockResolvedValue([]);
+    mockNotificationRepo.createQueryBuilder.mockReturnValue(mockQuery);
+    mockNotificationSubmissionRepo.createQueryBuilder.mockReturnValue(
+      mockQuery,
     );
 
-    const result = await service.search(mockSearchDto);
+    const result = await service.search(mockSearchDto, {} as any);
```

Comment:
is there a way not to use as any?

Reply:
Done! Used a mocked queryRunner instead.